### ISSUE TITLE
[no-implied-eval] False Positive Fired with Binded Class Func

### ISSUE THREAD

**Issue by gfyoung** [typescript-eslint/typescript-eslint#3582]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-implied-eval": "error"
  }
}
```

```TS
class Foo {
  name: string;

  constructor() {
    this.name = 'foo';
  }

  func1(): string {
    return this.name;
  }

  func2(): void {
    setTimeout(this.func1.bind(this), 1);
  }
}

export default new Foo();
```

**Expected Result**

No errors

**Actual Result**

~~~
13:16  error  Implied eval. Consider passing a function  @typescript-eslint/no-implied-eval
~~~

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.28.0` |
| `@typescript-eslint/parser`        | `4.28.0` |
| `TypeScript`                       | `4.3.4` |
| `ESLint`                           | `7.29.0` |
| `node`                             | `16.3.0` |
